structure_tests/structure_test.go

```diff
@@ -71,29 +74,56 @@ func Parse(fp string) (StructureTest, error) {
 	if version == "" {
 		return nil, errors.New("Please provide JSON schema version")
 	}
-	st := schemaVersions[version]()
-	if st == nil {
+
+	var st StructureTest
+	if schemaVersion, ok := schemaVersions[version]; ok {
+		st = schemaVersion()
+	} else {
 		return nil, errors.New("Unsupported schema version: " + version)
 	}
 
 	unmarshal(testContents, st)
+
 	tests, ok := st.(StructureTest) //type assertion
 	if !ok {
 		return nil, errors.New("Error encountered when type casting Structure Test interface")
 	}
+	tests.SetDriver(driverImpl)
 	return tests, nil
 }
 
 var configFiles arrayFlags
+var imageName, driver string
+var driverImpl drivers.Driver
 
 func TestMain(m *testing.M) {
-	flag.Var(&configFiles, "config", "path to the .yaml file containing test definitions")
+	flag.StringVar(&imageName, "image", "", "path to test image")
+	flag.StringVar(&driver, "driver", "docker", "driver to use when running tests")
+
 	flag.Parse()
+	configFiles = flag.Args()
+	stdout := bufio.NewWriter(os.Stdout)
```

Comment:
You should be able to use fmt.Println() in most of hte cases here instead of writing to stdout and flushing manually.

Reply:
the `testing` package in golang swallows all `fmt.Println` calls, so they won't appear in stdout like normal here. looks like I can use `fmt.Fprintln` here and pass it stdout though, which is better than writing and flushing manually

Reply:
Ah, you probably want t.Log* then. It's toggleable with the go testing flags.

Reply:
problem is, I'm not in the test run so I don't have access to a `t` yet...